core/src/consensus.rs

```diff
@@ -187,18 +184,21 @@ pub fn valid_header_version(height: u64, version: HeaderVersion) -> bool {
 	height < 4 * HARD_FORK_INTERVAL && version == header_version(height)
 }
 
-/// Number of blocks used to calculate difficulty adjustments
-pub const DIFFICULTY_ADJUST_WINDOW: u64 = HOUR_HEIGHT;
+/// Number of blocks used to calculate difficulty adjustment by Damped Moving Average
+pub const DMA_WINDOW: u64 = HOUR_HEIGHT;
+
+/// Difficulty adjustment half life is 4 hours
+pub const WTEMA_HALF_LIFE: u64 = 4 * HOUR_SEC;
```

Comment:
How was `WTEMA_HALF_LIFE` parameter selection determined?

Reply:
it's a tradeoff between response speed and stability. the 240 blocks closely matches BTH's choice of the equivalent of 288 blocks.